agent/api/task/task.go

```diff
@@ -281,7 +283,7 @@ type Task struct {
 	// setIdOnce is used to set the value of this task's id only the first time GetID is invoked
 	setIdOnce sync.Once
 
-	ServiceConnectConfig *ServiceConnectConfig `json:"ServiceConnectConfig,omitempty"`
+	ServiceConnectConfig *serviceconnect.Config `json:"Config,omitempty"`
```

Comment:
I don't think we want to change the serialized key with this change.

Reply:
You are right this is rather unintended. Will fix.